src/shutdown.rs

```diff
@@ -175,7 +186,7 @@ impl SourceShutdownCoordinator {
     /// resolves once all sources have either shut down completely, or have been sent the
     /// force shutdown signal.  The force shutdown signal will be sent to any sources that
     /// don't cleanly shut down before the given `deadline`.
-    pub fn shutdown_all(self, deadline: Instant) -> impl Future<Item = (), Error = ()> {
+    pub fn shutdown_all(self, deadline: Instant) -> impl Future01<Item = (), Error = ()> {
```

Comment:
The changed behavior of `let force_shutdown_tripwire = force_shutdown_tripwire.then(tripwire_handler);` changes the behavior of this returned `Future`, so that should be documented with something like: "Dropping this future prematurely will force shutdown the remaining sources."